server/protocols/jmap-rfc-8621/src/main/scala/org/apache/james/jmap/mail/Email.scala

```diff
@@ -82,25 +82,24 @@ object Email {
       mime4JMessage <- Try(defaultMessageBuilder.parseMessage(firstMessage.getFullContent.getInputStream))
       bodyStructure <- EmailBodyPart.of(messageId, mime4JMessage)
     } yield {
-      Email(
-        id = messageId,
-        size = sanitizeSize(firstMessage.getSize),
-        bodyStructure = bodyStructure,
-        textBody = bodyStructure.textBody,
-        htmlBody = bodyStructure.htmlBody,
-        attachments = bodyStructure.attachments,
-        headers = asEmailHeaders(mime4JMessage.getHeader),
-        headerMessageId = asEmailHeaderValue(mime4JMessage.getHeader.getField("MessageId")),
-        headerSubject = asEmailHeaderValue(mime4JMessage.getHeader.getField("Subject")),
-        headerFrom = asEmailHeaderValue(mime4JMessage.getHeader.getField("From")),
-        headerTo = asEmailHeaderValue(mime4JMessage.getHeader.getField("To")),
-        headerCc = asEmailHeaderValue(mime4JMessage.getHeader.getField("Cc")),
-        headerBcc = asEmailHeaderValue(mime4JMessage.getHeader.getField("Bcc")),
-        headerReplyTo = asEmailHeaderValue(mime4JMessage.getHeader.getField("ReplyTo")),
-        headerInReplyTo = asEmailHeaderValue(mime4JMessage.getHeader.getField("InReplyTo")),
-        headerSender = asEmailHeaderValue(mime4JMessage.getHeader.getField("Sender")),
-        headerReferences = asEmailHeaderValue(mime4JMessage.getHeader.getField("References")),
-        headerSentAt = asEmailHeaderValue(mime4JMessage.getHeader.getField("SentAt")))
+      Email(EmailMetadata(id = messageId,
+        size = sanitizeSize(firstMessage.getSize)),
+        EmailHeaders(headers = asEmailHeaders(mime4JMessage.getHeader)),
+        EmailBody(bodyStructure = bodyStructure,
+          textBody = bodyStructure.textBody,
+          htmlBody = bodyStructure.htmlBody,
+          attachments = bodyStructure.attachments),
+        Map(("header:MessageId", asEmailHeaderValue(mime4JMessage.getHeader.getField("MessageId"))),
+          ("header:Subject", asEmailHeaderValue(mime4JMessage.getHeader.getField("Subject"))),
+          ("header:From", asEmailHeaderValue(mime4JMessage.getHeader.getField("From"))),
+          ("header:To", asEmailHeaderValue(mime4JMessage.getHeader.getField("To"))),
+          ("header:Cc", asEmailHeaderValue(mime4JMessage.getHeader.getField("Cc"))),
+          ("header:Bcc", asEmailHeaderValue(mime4JMessage.getHeader.getField("Bcc"))),
+          ("header:ReplyTo", asEmailHeaderValue(mime4JMessage.getHeader.getField("ReplyTo"))),
+          ("header:InReplyTo", asEmailHeaderValue(mime4JMessage.getHeader.getField("InReplyTo"))),
+          ("header:Sender", asEmailHeaderValue(mime4JMessage.getHeader.getField("Sender"))),
+          ("header:References", asEmailHeaderValue(mime4JMessage.getHeader.getField("References"))),
+          ("header:SentAt", asEmailHeaderValue(mime4JMessage.getHeader.getField("SentAt")))))
```

Comment:
Can the content of this map be dynamically generated from the content of the property object?